packages/@sanity/form-builder/src/inputs/ArrayInput/ArrayFunctions.tsx

```diff
@@ -29,6 +32,25 @@ export default class ArrayFunctions extends React.Component<Props, {}> {
     const item = onCreateValue(type)
     onAppendItem(item)
   }
+  handleDuplicateBtnClick = (menuItem: {item: Record<string, any>}) => {
+    this.handleDuplicateItem(menuItem.item)
+  }
+  handleDuplicateItem = item => {
+    const {onAppendItem} = this.props
+    const newItem = item.hasOwnProperty('_key') ? updateKeys(Object.assign({}, item), '_key') : item
+    onAppendItem(newItem)
+  }
+  renderSelectDuplicate() {
+    const items = this.props.value.map(item => ({
+      title: item.title || item.name || item.where,
```

Comment:
This is the trickiest part of the whole thing - an object's name could be anything depending on the schema... At the moment, it contains what is most likely _and_ something that renders in the test-studio